src/Symfony/Component/Security/Core/User/UserProviderInterface.php

```diff
@@ -41,6 +43,8 @@ interface UserProviderInterface
      *
      * @return UserInterface
      *
+     * @psalm-return TUser
```

Comment:
we can use `@return` for this, we do already in some places IIRC

Reply:
It's indeed used in some place, because the @return was not necessary (the native type was added).
For this method, 
```
public function refreshUser(UserInterface $user);
```
there is no native return type.

`@return TUser` could loose some IDE or developer which doesn't know generics.
that's why I thought about adding the both annotations.

Do you prefer I change the `@return UserInterface` to `@return TUser` ?

Reply:
> Do you prefer I change the @return UserInterface to @return TUser ?
 
I do yes

Reply:
Done in https://github.com/symfony/symfony/pull/50399/commits/10cd4cc307447162c70776ccd24cec7cebd23708

Reply:
When we don't have the native return type yet, we need to keep `@return` with the future return type. Otherwise we break things for the DebugClassLoader with its deprecation warning about the future return type and its type patching.

Reply:
Reverted then